src/core/server/core_app/core_app.ts

```diff
@@ -7,27 +7,32 @@
  */
 
 import Path from 'path';
-import { fromRoot } from '../../../core/server/utils';
+import { Env } from '@kbn/config';
 
+import { fromRoot } from '../utils';
 import { InternalCoreSetup } from '../internal_types';
 import { CoreContext } from '../core_context';
 import { Logger } from '../logging';
+import { registerBundleRoutes } from './bundle_routes';
+import { UiPlugins } from '../plugins';
 
 /** @internal */
 export class CoreApp {
   private readonly logger: Logger;
+  private readonly env: Env;
 
   constructor(core: CoreContext) {
     this.logger = core.logger.get('core-app');
+    this.env = core.env;
   }
 
-  setup(coreSetup: InternalCoreSetup) {
+  setup(coreSetup: InternalCoreSetup, uiPlugins: UiPlugins) {
```

Comment:
We need the `uiPlugins` definition to register the bundle routes, so I had to add it to the `CoreApp.setup` parameters, as it's not available from the setup contract.